test/e2e/flask/multichain-api/non-evm/wallet_createSession.spec.ts

```diff
@@ -0,0 +1,106 @@
+    it('should only select the specified scopes requested by the user', async function () {
+      await withSolanaAccountSnap(
+        {
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+          title: this.test?.fullTitle(),
+        },
+        async (driver, _, extensionId) => {
+          const requestScopesToNetworkMap = {
+            'eip155:1': 'Ethereum Mainnet',
+            [SOLANA_SCOPE]: 'Solana',
+          };
+
+          const requestScopes = Object.keys(requestScopesToNetworkMap);
+          const networksToRequest = Object.values(requestScopesToNetworkMap);
+
+          const testDapp = new TestDappMultichain(driver);
+          await testDapp.openTestDappPage();
+          await testDapp.connectExternallyConnectable(extensionId);
+          await testDapp.initCreateSessionScopes(requestScopes);
+
+          // navigate to network selection screen
+          const permissionsTab = await driver.findElement(
+            '[data-testid="permissions-tab"]',
+          );
+          await permissionsTab.click();
+          const editButtons = await driver.findElements('[data-testid="edit"]');
+          await editButtons[1].click();
+          await driver.delay(largeDelayMs);
+
+          const networkListItems = await driver.findElements(
+            '.multichain-network-list-item',
+          );
+
+          for (const item of networkListItems) {
+            const networkNameDiv = await item.findElement(
+              By.css('div[data-testid]'),
+            );
+            const network = await networkNameDiv.getAttribute('data-testid');
+
+            const checkbox = await item.findElement(
+              By.css('input[type="checkbox"]'),
+            );
+            const isChecked = await checkbox.isSelected();
+            if (networksToRequest.includes(network)) {
+              assert.strictEqual(
+                isChecked,
+                true,
+                `Expected ${network} to be selected.`,
+              );
+            } else {
+              assert.strictEqual(
+                isChecked,
+                false,
+                `Expected ${network} to NOT be selected.`,
+              );
+            }
+          }
+        },
+      );
+    });
+  });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_createSession` with Solana scope, without any accounts requested', function () {
+    it('should automatically select the current active Solana account', async function () {
+      await withSolanaAccountSnap(
+        {
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+          title: this.test?.fullTitle(),
+        },
+        async (driver, _, extensionId) => {
+          const testDapp = new TestDappMultichain(driver);
+          await testDapp.openTestDappPage();
+          await testDapp.connectExternallyConnectable(extensionId);
+          await testDapp.initCreateSessionScopes([SOLANA_SCOPE]);
```

Comment:
I'm noticing that in the test we add the Solana scope manually, we shouldn't need to do that in this case since there is a checkbox for solana mainnet

Reply:
No this is fine for now